pvlib/windspeed.py

```diff
@@ -0,0 +1,165 @@
+
+    Raises
+    ------
+    ValueError
+        If neither of ``exponent`` nor a ``surface_type`` is given.
+        If both ``exponent`` nor a ``surface_type`` is given. These parameters
+        are mutually exclusive.
+
+    KeyError
+        If the specified ``surface_type`` is invalid.
+
+    Notes
+    -----
+    The equation for calculating the wind speed at a height of :math:`h` is
+    given by the following power law equation [1]_, [2]_:
+
+    .. math::
+       :label: wind speed
+
+        windspeed_h = windspeed_{href} \cdot \left( \frac{h}{h_{ref}} \right)^a
+
+    where :math:`h` [m] is the height at which we would like to calculate the
+    wind speed, :math:`h_{ref}` [m] is the reference height at which the wind
+    speed is known, and :math:`windspeed_h` [m/s] and :math:`windspeed_{href}`
+    [m/s] are the corresponding wind speeds at these heights. :math:`a` is a
+    value that depends on the surface type. Some values found in the literature
+    [1]_ for :math:`a` are the following:
+
+    .. table:: Values for the Hellmann-exponent
+
+    +-----------+--------------------+------------------+------------------+
+    | Stability | Open water surface | Flat, open coast | Cities, villages |
+    +===========+====================+==================+==================+
+    | Unstable  | 0.06               | 0.10             | 0.27             |
+    +-----------+--------------------+------------------+------------------+
+    | Neutral   | 0.11               | 0.16             | 0.40             |
+    +-----------+--------------------+------------------+------------------+
+    | Stable    | 0.27               | 0.34             | 0.60             |
+    +-----------+--------------------+------------------+------------------+
+
+    In a report by SANDIA [3]_, this equation was experimentally tested for a
+    height of 30 ft (9.144 m) and the following coefficients were recommended:
+    :math:`h_{ref} = 9.144` [m], :math:`a = 0.219` [-], and
+    :math:`windspeed_{href}` is the wind speed at a height of 9.144 [m].
+
+    It should be noted that the equation retuns a value of NaN if the
+    calculated wind speed is negative or a complex number.
+
+    Warning
+    -------
+    Module temperature functions often require wind speeds at a heigt of 10 m
+    and not the wind speed at the module height.
+
+    For example, the following temperature functions require the input wind
+    speed to be 10 m: :py:func:`~pvlib.temperature.sapm_cell`,
+    :py:func:`~pvlib.temperature.sapm_module`, and
+    :py:func:`~pvlib.temperature.generic_linear`, whereas the
+    :py:func:`~pvlib.temperature.fuentes` model requires wind speed at 9.144 m.
+
+    Additionally, the heat loss coefficients of some models have been developed
+    for wind speed measurements at 10 m (e.g.,
+    :py:func:`~pvlib.temperature.pvsyst_cell`,
+    :py:func:`~pvlib.temperature.faiman`, and
+    :py:func:`~pvlib.temperature.faiman_rad`).
```

Comment:
Thoughts on removing this from this function and adding it somewhere else @AdamRJensen and @kandersolar?

Reply:
I think having a dedicated page would be nice in the future, but to avoid dragging this further out I say let's keep it as is, and then we can always move it.